### PR TITLE
Removes un-needed saves

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/9959

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- make sure that wallet is created correctly

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by NejcZdovc** [comment]
CI failed on android test timeout, but tests passed so CI is good

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc [hunk 1]
```diff
@@ -23,56 +23,39 @@ using std::placeholders::_1;
...
-  ledger_->LoadURL(url, {}, "", "", ledger::UrlMethod::GET, req_callback);
-}
-
-std::string Create::GetAnonizeProof(const std::string& registrarVK,
-                                    const std::string& id,
-                                    std::string* preFlight) {
+std::string GetAnonizeProof(
+    const std::string& registrarVK,
+    const std::string& id,
+    std::string* preFlight) {
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:31]: Maybe add a `DCHECK(preFlight)` here?

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc [hunk 2]
```diff
@@ -23,56 +23,39 @@ using std::placeholders::_1;
...
   const char* cred = makeCred(id.c_str());
-  if (cred != nullptr) {
-    *preFlight = cred;
-    // should fix in
-    // https://github.com/brave-intl/bat-native-anonize/issues/11
-    free((void*)cred); // NOLINT
-  } else {
+  if (cred == nullptr) {
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:33]: Nit: `!cred` is a little more idiomatic.

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc [hunk 3]
```diff
@@ -23,56 +23,39 @@ using std::placeholders::_1;
...
-    proof = proofTemp;
-    // should fix in
-    // https://github.com/brave-intl/bat-native-anonize/issues/11
-    free((void*)proofTemp); // NOLINT
-  } else {
+
+  *preFlight = cred;
+  // should fix in
+  // https://github.com/brave-intl/bat-native-anonize/issues/11
+  free((void*)cred); // NOLINT
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:40]: Just double-checking: this was allocated with `malloc` and not `new`?

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc [hunk 4]
```diff
@@ -23,56 +23,39 @@ using std::placeholders::_1;
...
+
+  const char* proof_temp = registerUserMessage(
+      preFlight->c_str(),
+      registrarVK.c_str());
+
+  if (proof_temp == nullptr) {
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:46]: Could be `!proof_temp` if you want.

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:31]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc (hunk 2 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:33]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc (hunk 4 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:46]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc (hunk 3 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:40]: not sure about it as code was just moved, no code/logic change

Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:42]: No worries, I checked and the memory is allocated via `srtdup` so `free` is correct.

**Pull request merged**